enterprise/internal/campaigns/webhooks_gitlab.go

```diff
@@ -173,20 +168,26 @@ func (h *GitLabWebhook) handleEvent(ctx context.Context, extSvc *types.ExternalS
 	case *webhooks.MergeRequestApprovedEvent,
 		*webhooks.MergeRequestUnapprovedEvent,
 		*webhooks.MergeRequestUpdateEvent:
-		if err := h.enqueueChangesetSyncFromEvent(ctx, esID, e.(webhooks.MergeRequestEventContainer).ToEvent()); err != nil {
+		if err := h.enqueueChangesetSyncFromEvent(ctx, esID, e.(webhooks.MergeRequestEventCommonContainer).ToEventCommon()); err != nil {
 			return &httpError{
 				code: http.StatusInternalServerError,
 				err:  err,
 			}
 		}
 		return nil
 
-	// All other merge request events are state events.
-	case stateMergeRequestEvent:
-		if err := h.handleMergeRequestStateEvent(ctx, esID, e); err != nil {
+	case *webhooks.MergeRequestCloseEvent,
+		*webhooks.MergeRequestMergeEvent,
+		*webhooks.MergeRequestReopenEvent,
+		*webhooks.MergeRequestDraftEvent,
+		*webhooks.MergeRequestUndraftEvent:
+		eventCommon := e.(webhooks.MergeRequestEventCommonContainer).ToEventCommon()
+		event := e.(webhooks.MergeRequestEventToEvent).ToEvent()
```

Comment:
Does it make sense to define an interface that combines these two so you can simplify the `case` here and don't need the casts?

Reply:
Might be, I wanted to be explicit though so we know which events we can upsert and which we can't. 🤔 WDYT?

Reply:
Not sure I understand or whether I've missed something in the first place, but what I meant was: can't you do something like this?

```go
type resourceStateEvent interface {
	webhooks.MergeRequestEventCommonContainer
	webhooks.MergeRequestEventToEvent
}
```

and then here, you can do:

```suggestion
		eventCommon := e.ToEventCommon()
		event := e.ToEvent()
```

Reply:
Ah I misunderstood you then 🤦 That should work!